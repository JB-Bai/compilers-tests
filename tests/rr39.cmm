float CpWeuF(){
	int DogoCjJ;
	struct bcRKpF wKJ8;
	while(jf5CCjht1cu6 / _pek0CSr_Yc && 0xF8) if(OZ2TlB4E3k) 0127;
	29.7415 == PqgXQpia;
}
float PwMaiv6bZ(){
	int u0h_xJ = 159998, nzTa[0xeE06] = 30705 * 054.1556E+6 = EoEfMh3(ZaGXErOfxY == 2.11.jchdJGQ) - ! 3.50E0 > 96.69 * ! _LXnv7R((926.34.Z7S7ENROls - ! ! XMHrs() && u_8yu7Fu(827.110 * qkxhpH2()[54][kjObT() - j2MFWy] + 92.879.GC8Y7V / S1xyCF2Fls >= KnvimYk <= Nre4XSp || 0xA / 0xC1A7 && yZMuAvq = ! - JgbQdbP0 * ! - CariqlfXN || szvmzh(), WNLImlVbq(- 05541 >=(oBZ5MUi) && gIOwycy(0xa8 && 58396[62.47e0] = 428, sITmegwM, !(UL4aFGw) && X6zSIYH9 - 29346 / ItMYz || jzaCDHc6), BIe / 905469, - 0xB0B.yPbe && ZryMzo7Sy.XLlHt4Q / 0x6005.QID9qsgf || 469.0.PJnfcQU, B8yqDy2bJy / 0xa9 *(OTpKqEL5) != 18.803E4), 66.03 * vuypBWk) = 436.4e-7 && j47E2lF * KFXSETl5() - 0xD6 + 0xE76d[0x55A] || 79.054 - 0x10 || 436.3805e-7 - AB9EMX - 055452 >= ! 0xE1 / PnZc2_a[p_hW8tsK || Y3UazAJ_] * 9216.82E0.A6Ynk7q + RxKhNaMO && 63.331E5)) < ! oz8oY9Q / 8987.84 || 66.7682 || 0x3d40e || AkqG5e().hnpAps4P, cd2iu = ! 652.08 / 5.174 || A34Y0fJj.NxSnT_c2, qlBPWpq;
	XcgfNW14(FovcudY());
	VB6aO132yQ;
	odxweXBSU - 10320;
	if(y05f0vPz9n) ! ff6AGfTo.UZcNp3bQdw;
	else 572.398;
	if(0xaf - 4.78E-9 + 5.9E9 * tE8xIDqrd3(0xfc0)) QeKFhx >= 550.91;
	if(848.178e-6 / 0331 + bXrdSU1X() + 187.5959 / uw0nPP(w8MNXw7OEjt(AlmxMHgq), xWYl83U2I, 015376[_ijxhy0 = _VpKAknSG +(10.08)]) &&(QxMX = 0xBe[11.607 +(q4yPEEhX2Y().o12BSJd)]) &&(89774 || rtmt3pSc2v()) = _KWSuz1() - ! 167.52 && Ms2oumfHVC[182.69 && 07061] * wGiw4NgU - 99.69E+9) return 4673;
	else while(0xa51) return DVeTB0zkE;
	(72.643 && ZRrdZm() / 0xCF9A);
	8.826e+1;
	15113 = R2Aqh() - rqVDeHeZ2e() = vRVTB561();
	9.7299;
	0x480A / 025 - 033742 * 6671.2098 >= 819[6.803 && 0xFDa];
}
float;

