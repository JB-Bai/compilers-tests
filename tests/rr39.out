Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      FunDec (1)
        ID: CpWeuF
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: DogoCjJ
            SEMI
          DefList (3)
            Def (3)
              Specifier (3)
                StructSpecifier (3)
                  STRUCT
                  Tag (3)
                    ID: bcRKpF
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: wKJ8
              SEMI
        StmtList (4)
          Stmt (4)
            WHILE
            LP
            Exp (4)
              Exp (4)
                Exp (4)
                  ID: jf5CCjht1cu6
                DIV
                Exp (4)
                  ID: _pek0CSr_Yc
              AND
              Exp (4)
                INT: 248
            RP
            Stmt (4)
              IF
              LP
              Exp (4)
                ID: OZ2TlB4E3k
              RP
              Stmt (4)
                Exp (4)
                  INT: 87
                SEMI
          StmtList (5)
            Stmt (5)
              Exp (5)
                Exp (5)
                  FLOAT: 29.741501
                RELOP
                Exp (5)
                  ID: PqgXQpia
              SEMI
        RC
    ExtDefList (7)
      ExtDef (7)
        Specifier (7)
          TYPE: float
        FunDec (7)
          ID: PwMaiv6bZ
          LP
          RP
        CompSt (7)
          LC
          DefList (8)
            Def (8)
              Specifier (8)
                TYPE: int
              DecList (8)
                Dec (8)
                  VarDec (8)
                    ID: u0h_xJ
                  ASSIGNOP
                  Exp (8)
                    INT: 159998
                COMMA
                DecList (8)
                  Dec (8)
                    VarDec (8)
                      VarDec (8)
                        ID: nzTa
                      LB
                      INT: 60934
                      RB
                    ASSIGNOP
                    Exp (8)
                      Exp (8)
                        Exp (8)
                          INT: 30705
                        STAR
                        Exp (8)
                          FLOAT: 54155600.000000
                      ASSIGNOP
                      Exp (8)
                        Exp (8)
                          Exp (8)
                            Exp (8)
                              Exp (8)
                                Exp (8)
                                  Exp (8)
                                    ID: EoEfMh3
                                    LP
                                    Args (8)
                                      Exp (8)
                                        Exp (8)
                                          ID: ZaGXErOfxY
                                        RELOP
                                        Exp (8)
                                          Exp (8)
                                            FLOAT: 2.110000
                                          DOT
                                          ID: jchdJGQ
                                    RP
                                  MINUS
                                  Exp (8)
                                    NOT
                                    Exp (8)
                                      FLOAT: 3.500000
                                RELOP
                                Exp (8)
                                  Exp (8)
                                    FLOAT: 96.690002
                                  STAR
                                  Exp (8)
                                    NOT
                                    Exp (8)
                                      ID: _LXnv7R
                                      LP
                                      Args (8)
                                        Exp (8)
                                          LP
                                          Exp (8)
                                            Exp (8)
                                              Exp (8)
                                                Exp (8)
                                                  Exp (8)
                                                    FLOAT: 926.340027
                                                  DOT
                                                  ID: Z7S7ENROls
                                                MINUS
                                                Exp (8)
                                                  NOT
                                                  Exp (8)
                                                    NOT
                                                    Exp (8)
                                                      ID: XMHrs
                                                      LP
                                                      RP
                                              AND
                                              Exp (8)
                                                ID: u_8yu7Fu
                                                LP
                                                Args (8)
                                                  Exp (8)
                                                    Exp (8)
                                                      Exp (8)
                                                        Exp (8)
                                                          Exp (8)
                                                            Exp (8)
                                                              Exp (8)
                                                                FLOAT: 827.109985
                                                              STAR
                                                              Exp (8)
                                                                Exp (8)
                                                                  Exp (8)
                                                                    ID: qkxhpH2
                                                                    LP
                                                                    RP
                                                                  LB
                                                                  Exp (8)
                                                                    INT: 54
                                                                  RB
                                                                LB
                                                                Exp (8)
                                                                  Exp (8)
                                                                    ID: kjObT
                                                                    LP
                                                                    RP
                                                                  MINUS
                                                                  Exp (8)
                                                                    ID: j2MFWy
                                                                RB
                                                            PLUS
                                                            Exp (8)
                                                              Exp (8)
                                                                Exp (8)
                                                                  FLOAT: 92.878998
                                                                DOT
                                                                ID: GC8Y7V
                                                              DIV
                                                              Exp (8)
                                                                ID: S1xyCF2Fls
                                                          RELOP
                                                          Exp (8)
                                                            ID: KnvimYk
                                                        RELOP
                                                        Exp (8)
                                                          ID: Nre4XSp
                                                      OR
                                                      Exp (8)
                                                        Exp (8)
                                                          Exp (8)
                                                            INT: 10
                                                          DIV
                                                          Exp (8)
                                                            INT: 49575
                                                        AND
                                                        Exp (8)
                                                          ID: yZMuAvq
                                                    ASSIGNOP
                                                    Exp (8)
                                                      Exp (8)
                                                        Exp (8)
                                                          NOT
                                                          Exp (8)
                                                            MINUS
                                                            Exp (8)
                                                              ID: JgbQdbP0
                                                        STAR
                                                        Exp (8)
                                                          NOT
                                                          Exp (8)
                                                            MINUS
                                                            Exp (8)
                                                              ID: CariqlfXN
                                                      OR
                                                      Exp (8)
                                                        ID: szvmzh
                                                        LP
                                                        RP
                                                  COMMA
                                                  Args (8)
                                                    Exp (8)
                                                      ID: WNLImlVbq
                                                      LP
                                                      Args (8)
                                                        Exp (8)
                                                          Exp (8)
                                                            Exp (8)
                                                              MINUS
                                                              Exp (8)
                                                                INT: 2913
                                                            RELOP
                                                            Exp (8)
                                                              LP
                                                              Exp (8)
                                                                ID: oBZ5MUi
                                                              RP
                                                          AND
                                                          Exp (8)
                                                            ID: gIOwycy
                                                            LP
                                                            Args (8)
                                                              Exp (8)
                                                                Exp (8)
                                                                  Exp (8)
                                                                    INT: 168
                                                                  AND
                                                                  Exp (8)
                                                                    Exp (8)
                                                                      INT: 58396
                                                                    LB
                                                                    Exp (8)
                                                                      FLOAT: 62.470001
                                                                    RB
                                                                ASSIGNOP
                                                                Exp (8)
                                                                  INT: 428
                                                              COMMA
                                                              Args (8)
                                                                Exp (8)
                                                                  ID: sITmegwM
                                                                COMMA
                                                                Args (8)
                                                                  Exp (8)
                                                                    Exp (8)
                                                                      Exp (8)
                                                                        NOT
                                                                        Exp (8)
                                                                          LP
                                                                          Exp (8)
                                                                            ID: UL4aFGw
                                                                          RP
                                                                      AND
                                                                      Exp (8)
                                                                        Exp (8)
                                                                          ID: X6zSIYH9
                                                                        MINUS
                                                                        Exp (8)
                                                                          Exp (8)
                                                                            INT: 29346
                                                                          DIV
                                                                          Exp (8)
                                                                            ID: ItMYz
                                                                    OR
                                                                    Exp (8)
                                                                      ID: jzaCDHc6
                                                            RP
                                                        COMMA
                                                        Args (8)
                                                          Exp (8)
                                                            Exp (8)
                                                              ID: BIe
                                                            DIV
                                                            Exp (8)
                                                              INT: 905469
                                                          COMMA
                                                          Args (8)
                                                            Exp (8)
                                                              Exp (8)
                                                                Exp (8)
                                                                  MINUS
                                                                  Exp (8)
                                                                    Exp (8)
                                                                      INT: 2827
                                                                    DOT
                                                                    ID: yPbe
                                                                AND
                                                                Exp (8)
                                                                  Exp (8)
                                                                    Exp (8)
                                                                      ID: ZryMzo7Sy
                                                                    DOT
                                                                    ID: XLlHt4Q
                                                                  DIV
                                                                  Exp (8)
                                                                    Exp (8)
                                                                      INT: 24581
                                                                    DOT
                                                                    ID: QID9qsgf
                                                              OR
                                                              Exp (8)
                                                                Exp (8)
                                                                  FLOAT: 469.000000
                                                                DOT
                                                                ID: PJnfcQU
                                                            COMMA
                                                            Args (8)
                                                              Exp (8)
                                                                Exp (8)
                                                                  Exp (8)
                                                                    Exp (8)
                                                                      ID: B8yqDy2bJy
                                                                    DIV
                                                                    Exp (8)
                                                                      INT: 169
                                                                  STAR
                                                                  Exp (8)
                                                                    LP
                                                                    Exp (8)
                                                                      ID: OTpKqEL5
                                                                    RP
                                                                RELOP
                                                                Exp (8)
                                                                  FLOAT: 188030.000000
                                                      RP
                                                    COMMA
                                                    Args (8)
                                                      Exp (8)
                                                        Exp (8)
                                                          FLOAT: 66.029999
                                                        STAR
                                                        Exp (8)
                                                          ID: vuypBWk
                                                RP
                                            ASSIGNOP
                                            Exp (8)
                                              Exp (8)
                                                Exp (8)
                                                  Exp (8)
                                                    FLOAT: 0.000044
                                                  AND
                                                  Exp (8)
                                                    Exp (8)
                                                      Exp (8)
                                                        Exp (8)
                                                          ID: j47E2lF
                                                        STAR
                                                        Exp (8)
                                                          ID: KFXSETl5
                                                          LP
                                                          RP
                                                      MINUS
                                                      Exp (8)
                                                        INT: 214
                                                    PLUS
                                                    Exp (8)
                                                      Exp (8)
                                                        INT: 59245
                                                      LB
                                                      Exp (8)
                                                        INT: 1370
                                                      RB
                                                OR
                                                Exp (8)
                                                  Exp (8)
                                                    FLOAT: 79.054001
                                                  MINUS
                                                  Exp (8)
                                                    INT: 16
                                              OR
                                              Exp (8)
                                                Exp (8)
                                                  Exp (8)
                                                    Exp (8)
                                                      Exp (8)
                                                        FLOAT: 0.000044
                                                      MINUS
                                                      Exp (8)
                                                        ID: AB9EMX
                                                    MINUS
                                                    Exp (8)
                                                      INT: 23338
                                                  RELOP
                                                  Exp (8)
                                                    Exp (8)
                                                      Exp (8)
                                                        Exp (8)
                                                          NOT
                                                          Exp (8)
                                                            INT: 225
                                                        DIV
                                                        Exp (8)
                                                          Exp (8)
                                                            ID: PnZc2_a
                                                          LB
                                                          Exp (8)
                                                            Exp (8)
                                                              ID: p_hW8tsK
                                                            OR
                                                            Exp (8)
                                                              ID: Y3UazAJ_
                                                          RB
                                                      STAR
                                                      Exp (8)
                                                        Exp (8)
                                                          FLOAT: 9216.820312
                                                        DOT
                                                        ID: A6Ynk7q
                                                    PLUS
                                                    Exp (8)
                                                      ID: RxKhNaMO
                                                AND
                                                Exp (8)
                                                  FLOAT: 6333100.000000
                                          RP
                                      RP
                              RELOP
                              Exp (8)
                                Exp (8)
                                  NOT
                                  Exp (8)
                                    ID: oz8oY9Q
                                DIV
                                Exp (8)
                                  FLOAT: 8987.839844
                            OR
                            Exp (8)
                              FLOAT: 66.768204
                          OR
                          Exp (8)
                            INT: 250894
                        OR
                        Exp (8)
                          Exp (8)
                            ID: AkqG5e
                            LP
                            RP
                          DOT
                          ID: hnpAps4P
                  COMMA
                  DecList (8)
                    Dec (8)
                      VarDec (8)
                        ID: cd2iu
                      ASSIGNOP
                      Exp (8)
                        Exp (8)
                          Exp (8)
                            NOT
                            Exp (8)
                              FLOAT: 652.080017
                          DIV
                          Exp (8)
                            FLOAT: 5.174000
                        OR
                        Exp (8)
                          Exp (8)
                            ID: A34Y0fJj
                          DOT
                          ID: NxSnT_c2
                    COMMA
                    DecList (8)
                      Dec (8)
                        VarDec (8)
                          ID: qlBPWpq
              SEMI
          StmtList (9)
            Stmt (9)
              Exp (9)
                ID: XcgfNW14
                LP
                Args (9)
                  Exp (9)
                    ID: FovcudY
                    LP
                    RP
                RP
              SEMI
            StmtList (10)
              Stmt (10)
                Exp (10)
                  ID: VB6aO132yQ
                SEMI
              StmtList (11)
                Stmt (11)
                  Exp (11)
                    Exp (11)
                      ID: odxweXBSU
                    MINUS
                    Exp (11)
                      INT: 10320
                  SEMI
                StmtList (12)
                  Stmt (12)
                    IF
                    LP
                    Exp (12)
                      ID: y05f0vPz9n
                    RP
                    Stmt (12)
                      Exp (12)
                        NOT
                        Exp (12)
                          Exp (12)
                            ID: ff6AGfTo
                          DOT
                          ID: UZcNp3bQdw
                      SEMI
                    ELSE
                    Stmt (13)
                      Exp (13)
                        FLOAT: 572.398010
                      SEMI
                  StmtList (14)
                    Stmt (14)
                      IF
                      LP
                      Exp (14)
                        Exp (14)
                          Exp (14)
                            INT: 175
                          MINUS
                          Exp (14)
                            FLOAT: 0.000000
                        PLUS
                        Exp (14)
                          Exp (14)
                            FLOAT: 5900000256.000000
                          STAR
                          Exp (14)
                            ID: tE8xIDqrd3
                            LP
                            Args (14)
                              Exp (14)
                                INT: 4032
                            RP
                      RP
                      Stmt (14)
                        Exp (14)
                          Exp (14)
                            ID: QeKFhx
                          RELOP
                          Exp (14)
                            FLOAT: 550.909973
                        SEMI
                    StmtList (15)
                      Stmt (15)
                        IF
                        LP
                        Exp (15)
                          Exp (15)
                            Exp (15)
                              Exp (15)
                                Exp (15)
                                  Exp (15)
                                    Exp (15)
                                      FLOAT: 0.000848
                                    DIV
                                    Exp (15)
                                      INT: 217
                                  PLUS
                                  Exp (15)
                                    ID: bXrdSU1X
                                    LP
                                    RP
                                PLUS
                                Exp (15)
                                  Exp (15)
                                    FLOAT: 187.595901
                                  DIV
                                  Exp (15)
                                    ID: uw0nPP
                                    LP
                                    Args (15)
                                      Exp (15)
                                        ID: w8MNXw7OEjt
                                        LP
                                        Args (15)
                                          Exp (15)
                                            ID: AlmxMHgq
                                        RP
                                      COMMA
                                      Args (15)
                                        Exp (15)
                                          ID: xWYl83U2I
                                        COMMA
                                        Args (15)
                                          Exp (15)
                                            Exp (15)
                                              INT: 6910
                                            LB
                                            Exp (15)
                                              Exp (15)
                                                ID: _ijxhy0
                                              ASSIGNOP
                                              Exp (15)
                                                Exp (15)
                                                  ID: _VpKAknSG
                                                PLUS
                                                Exp (15)
                                                  LP
                                                  Exp (15)
                                                    FLOAT: 10.080000
                                                  RP
                                            RB
                                    RP
                              AND
                              Exp (15)
                                LP
                                Exp (15)
                                  Exp (15)
                                    ID: QxMX
                                  ASSIGNOP
                                  Exp (15)
                                    Exp (15)
                                      INT: 190
                                    LB
                                    Exp (15)
                                      Exp (15)
                                        FLOAT: 11.607000
                                      PLUS
                                      Exp (15)
                                        LP
                                        Exp (15)
                                          Exp (15)
                                            ID: q4yPEEhX2Y
                                            LP
                                            RP
                                          DOT
                                          ID: o12BSJd
                                        RP
                                    RB
                                RP
                            AND
                            Exp (15)
                              LP
                              Exp (15)
                                Exp (15)
                                  INT: 89774
                                OR
                                Exp (15)
                                  ID: rtmt3pSc2v
                                  LP
                                  RP
                              RP
                          ASSIGNOP
                          Exp (15)
                            Exp (15)
                              Exp (15)
                                ID: _KWSuz1
                                LP
                                RP
                              MINUS
                              Exp (15)
                                NOT
                                Exp (15)
                                  FLOAT: 167.520004
                            AND
                            Exp (15)
                              Exp (15)
                                Exp (15)
                                  Exp (15)
                                    ID: Ms2oumfHVC
                                  LB
                                  Exp (15)
                                    Exp (15)
                                      FLOAT: 182.690002
                                    AND
                                    Exp (15)
                                      INT: 3633
                                  RB
                                STAR
                                Exp (15)
                                  ID: wGiw4NgU
                              MINUS
                              Exp (15)
                                FLOAT: 99689996288.000000
                        RP
                        Stmt (15)
                          RETURN
                          Exp (15)
                            INT: 4673
                          SEMI
                        ELSE
                        Stmt (16)
                          WHILE
                          LP
                          Exp (16)
                            INT: 2641
                          RP
                          Stmt (16)
                            RETURN
                            Exp (16)
                              ID: DVeTB0zkE
                            SEMI
                      StmtList (17)
                        Stmt (17)
                          Exp (17)
                            LP
                            Exp (17)
                              Exp (17)
                                FLOAT: 72.642998
                              AND
                              Exp (17)
                                Exp (17)
                                  ID: ZRrdZm
                                  LP
                                  RP
                                DIV
                                Exp (17)
                                  INT: 53146
                            RP
                          SEMI
                        StmtList (18)
                          Stmt (18)
                            Exp (18)
                              FLOAT: 88.260002
                            SEMI
                          StmtList (19)
                            Stmt (19)
                              Exp (19)
                                Exp (19)
                                  INT: 15113
                                ASSIGNOP
                                Exp (19)
                                  Exp (19)
                                    Exp (19)
                                      ID: R2Aqh
                                      LP
                                      RP
                                    MINUS
                                    Exp (19)
                                      ID: rqVDeHeZ2e
                                      LP
                                      RP
                                  ASSIGNOP
                                  Exp (19)
                                    ID: vRVTB561
                                    LP
                                    RP
                              SEMI
                            StmtList (20)
                              Stmt (20)
                                Exp (20)
                                  FLOAT: 9.729900
                                SEMI
                              StmtList (21)
                                Stmt (21)
                                  Exp (21)
                                    Exp (21)
                                      Exp (21)
                                        Exp (21)
                                          INT: 18442
                                        DIV
                                        Exp (21)
                                          INT: 21
                                      MINUS
                                      Exp (21)
                                        Exp (21)
                                          INT: 14306
                                        STAR
                                        Exp (21)
                                          FLOAT: 6671.209961
                                    RELOP
                                    Exp (21)
                                      Exp (21)
                                        INT: 819
                                      LB
                                      Exp (21)
                                        Exp (21)
                                          FLOAT: 6.803000
                                        AND
                                        Exp (21)
                                          INT: 4058
                                      RB
                                  SEMI
          RC
      ExtDefList (23)
        ExtDef (23)
          Specifier (23)
            TYPE: float
          SEMI
